### PR TITLE
[CO-398] Fixing accounts specs

### PR DESCRIPTION
**Pull request by paweljakubas**
## Description

postTransaction test was adapted to
(i) the fact that in new data layer newly created accounts DO NOT have automatically created addresses - in contrast to old data layer. In the new data layer the invariant that an account comes with a fresh address only if we create a new wallet is enforced. It was done to accommodate frontend requirements of having a “fully usable” wallet upon creation without having invariants scattered all across the place.
(ii) test postTransaction not only to automatically created accounts but also to newly created accounts>

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CO-398

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [x] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## QA Steps
```
nix-build -A walletIntegrationTests -o launch_integration_tests
./launch_integration_tests --match "Accounts
```

### CODE REVIEW

**Comment by paweljakubas** [comment]
From https://hydra.iohk.io/build/260598/log:
```
Accounts
  can retrieve only an account's balance
    +++ OK, passed 1 tests.
  can retrieve only an account's addresses
    +++ OK, passed 1 tests.
  can retrieve initial and updated balances of several accounts from getAccountBalancesthat are equivalent to what is obtained from getAccounts
    +++ OK, passed 1 tests.
  redeeming avvm key gives rise to the corresponding increase of balance of wallet'account - mnemonic not used
    +++ OK, passed 1 tests.
```

File: wallet-new/integration/AccountSpecs.hs [hunk 1]
```diff
@@ -91,11 +106,14 @@ accountSpecs wRef wc =
...
+
             let addrAndAmount = zip (map (\(addr : _) -> addr) $ map accAddresses accs) amounts
-            forM_  addrAndAmount $ \(addr, amount) ->
+
+            etxnsResp <- forM addrAndAmount $ \(addr, amount) -> do
+                -- we have to wait until the previous transaction is accommodated in the blockchain
+                liftIO $ threadDelay 120000000
                 run $ postTransaction wc (payment amount addr)
 
-            liftIO $ threadDelay 90000000
```
Comment (KtorZ) [inline wallet-new/integration/AccountSpecs.hs]: By making a small change to the helper below:

```hs
    pollTransaction :: V1 Txp.TxId -> IO ()
    pollTransaction tid = do
        resp <- getTransactionIndex wc Nothing Nothing Nothing >>= shouldPrismFlipped _Right
        let Just tx = find ((== tid) . txId) (wrData resp)
        if (txStatus tx `elem` [InNewestBlocks, Persisted]) then
            return ()
        else
            threadDelay (5 * oneSecond) >> pollTransaction tid
```

We can just write this:

```hs
            run $ forM_ addrAndAmount \(addr, amount) -> do
                etxn <- postTransaction wc (payment amount addr)
                txn  <- fmap wrData etxn `shouldPrism` _Right
                ("waiting for Tx to be accepted", pollTransaction (txId txn)) `noLongerThan` (120 * oneSecond)
```

Which would greatly speed up the test.

**Comment by paweljakubas** [comment]
I will make this change in upcoming Transactions spec PR

**Comment by KtorZ** [comment]
Sweat. Note that this new data-layer is much faster to validate transactions than the legacy one. On my machine, it typically takes ~8 seconds for a transaction to get into a block even though it's not fully-confirmed, funds are available in the wallet. 
Note that I didn't experience any particular delay when fetching the index, but the helper should probably be adapted to the following to cope with this:

```hs
    pollTransaction :: V1 Txp.TxId -> IO ()
    pollTransaction tid = do
        resp <- getTransactionIndex wc Nothing Nothing Nothing >>= shouldPrismFlipped _Right
        let tx= find ((== tid) . txId) (wrData resp)
        if ((txStatus <$> tx) `elem` (Just <$> [InNewestBlocks, Persisted])) then
            return ()
        else
            threadDelay (5 * oneSecond) >> pollTransaction tid
```

**Pull request merged**